### PR TITLE
maglev: Perf related follow up items

### PR DESCRIPTION
**Pull request by brb**
Reviewable per commit.

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

File: pkg/maps/lbmap/lbmap.go [hunk 1]
```diff
@@ -40,10 +40,23 @@ var (
...
+	maglevTableSize        uint64
+}
+
+func New(maglev bool, maglevTableSize int) *LBBPFMap {
+	m := &LBBPFMap{}
 
-func New() *LBBPFMap {
-	return &LBBPFMap{}
+	if maglev {
+		m.maglevBackendIDsBuffer = make([]uint16, maglevTableSize)
```
Comment (aanm) [inline pkg/maps/lbmap/lbmap.go:55]: did you mean `make([]uint16, 0, maglevTableSize)`?

Comment (brb) [reply pkg/maps/lbmap/lbmap.go:55]: No, as we don't append to the buffer: https://github.com/cilium/cilium/pull/13431/files#diff-4ceea0eea54513281df37f713e9b1638R108

Comment (aditighag) [reply pkg/maps/lbmap/lbmap.go:55]: >lbmap.maglevBackendIDsBuffer[i] = backends[backendNames[pos]]

At L 97, elements are getting added to the slice by index and not by `append`. `make([]uint16, 0, maglevTableSize)` will create a slice of 0 length, and cause panic.

**Review by pchaigno — APPROVED** [review]

**Review by borkmann — APPROVED** [review]

**Pull request merged**